### ISSUE TITLE
Turn on dropout (add randomness) on inference

### ISSUE THREAD

**Issue by SergeyPetrakov** [manujosephv/pytorch_tabular#71]
How can we add randomness on inference for model FT-Transformer via dropout?

For example, we have code
```
model_config = FTTransformerConfig(
    task = "regression",
    learning_rate=1e-4,
    seed = 42,
    input_embed_dim = 32,
    num_heads = 16,
    num_attn_blocks = 10,
    ff_dropout = 0.2,
    out_ff_layers = "1024-512-256",
    out_ff_activation = "LeakyReLU",
    #out_ff_activation = "ReLU",
    out_ff_initialization="kaiming",
    batch_norm_continuous_input=False,
    #         target_range=[(df_train[col].min(),df_train[col].max()) for col in ['target']]
)



tabular_model_leakeyrelu = TabularModel(
    data_config=data_config,
    model_config=model_config,
    optimizer_config=optimizer_config,
    trainer_config=trainer_config
)

tabular_model_leakeyrelu.fit(train=df_train, validation=df_valid)

pred_df_leakeyrelu = tabular_model_leakeyrelu.predict(df_test, ret_logits=False)
pred_df_leakeyrelu.head()
```

**Comment by manujosephv** [comment]
if you are open to do a bit of coding, you can re-write the predict method. We place the model in `eval` mode before inference to turn off dropouts, batchnorm etc. But you can turn it back on by using the below function:

```python
def enable_dropout(model):
    """ Function to enable the dropout layers during test-time """
    for m in model.modules():
        if m.__class__.__name__.startswith('Dropout'):
            m.train()
```

So ideally., the flow should be:
model.eval() >> enable_dropout(model) >> rest of the predict function

I hope you are doing this to get the MC-dropout inference. Let me know if this works.. Maybe we can add this to the `TabularModel` as a PR as well.

**Comment by SergeyPetrakov** [comment]
Yes, your guess is correct, I am trying to do uncertainty estimation (one of the approach is MC-dropout, but also I'm trying ensemble approach), and I still can not receive distinguishable from model.eval() results for dropout inference, I even tried just to paste self.model.train() instead self.model.eval() inside prediction function

**Comment by SergeyPetrakov** [comment]
By the way, according to the uncertainty via ensemble. Basing on https://paperswithcode.com/paper/simple-and-scalable-predictive-uncertainty it is worth setting something like NLL as a loss function for regression that take on account both value and it's variance. For this it is necessary to have 2 outputs of the model (value and variance). Then I have to write my own custom class to FT-Transformer additionally, that has 2 outputs and such upgraded NLL loss or there is a simplier way to do this? May be it is possible to do this just using FTTransformerModel? Because now I'm trying to work with 
 FTTransformerConfig, but there I have no possibility to set 2 as output  dimension.
 
 Thank you for answering anyway, I hope my problem is clear)
